include/deal.II/differentiation/ad/ad_helpers.h

```diff
@@ -3280,6 +3280,379 @@ namespace Differentiation
+       * the vector field evaluated at the point defined by the independent
+       * variable values.
+       * @param[in] extractor_row An extractor associated with the input field
+       * variables. This effectively defines which components of the global set
+       * of dependent variables this field is associated with.
+       */
+      template <typename ExtractorType_Row>
+      typename internal::VectorFieldValue<dim, scalar_type, ExtractorType_Row>::
+        type
+        extract_value_component(const Vector<scalar_type> &values,
+                                const ExtractorType_Row &  extractor_row) const;
+
+      /**
+       * Extract the Jacobian of the subset of dependent functions
+       * $\mathbf{g} \subset \boldsymbol{\Psi}(\mathbf{X})$
+       * for a subset of independent variables
+       * $\mathbf{A} \subset \mathbf{X}$, i.e.
+       * @f[
+       *   \mathbf{J}(\mathbf{g})
+       *      = \frac{\partial\mathbf{g}(\mathbf{X})}{\partial\mathbf{A}}
+       * @f]
+       *
+       * @param[in] jacobian The Jacobian of the vector function with respect to
+       * all independent variables, i.e., that returned by compute_jacobian().
+       * @param[in] extractor_row An extractor associated with the input field
+       * variables for which the first index of the Jacobian is extracted.
+       * This effectively defines which components of the global set
+       * of the dependent variables that this field (representing a subset of
+       * dependent functions) is associated with.
+       * @param[in] extractor_col An extractor associated with the input field
+       * variables for which the second index of the Jacobian is extracted.
+       * This effectively defines which components of the global set
+       * of independent variables that this field (representing a subset of the
+       * independent variables) is associated with.
+       */
+      template <typename ExtractorType_Row, typename ExtractorType_Col>
+      typename internal::VectorFieldJacobian<dim,
+                                             scalar_type,
+                                             ExtractorType_Row,
+                                             ExtractorType_Col>::type
+      extract_jacobian_component(const FullMatrix<scalar_type> &jacobian,
+                                 const ExtractorType_Row &      extractor_row,
+                                 const ExtractorType_Col &extractor_col) const;
+
+      /**
+       * Extract the Jacobian of the subset of dependent functions
+       * $\mathbf{g} \subset \boldsymbol{\Psi}(\mathbf{X})$
+       * for a subset of independent variables
+       * $\mathbf{A} \subset \mathbf{X}$, i.e.
+       * @f[
+       *   \mathbf{J}(\mathbf{g})
+       *      = \frac{\partial\mathbf{g}(\mathbf{X})}{\partial\mathbf{A}}
+       * @f]
+       *
+       * This function is a specialization of the above for rank-0 tensors
+       * (scalars).
+       */
+      Tensor<0, dim, scalar_type>
```

Comment:
Isn't this just a scalar?

Reply:
I've made a note in #7849 to do this. I wanted everything to remain consistent (because the extractor helper classes declare  `Tensor<0, dim, scalar_type>` to be the jacobian type. I would want to change this consistently throughout these classes.

Reply:
As mentioned in #7849, I don't feel strongly about it. You make the call.